src/validators/union.rs

```diff
@@ -307,7 +307,33 @@ impl Validator for TaggedUnionValidator {
                                 if self.strict {
                                     value.strict_str()
                                 } else {
-                                    value.lax_str()
+                                    match value.laxer_str() {
```

Comment:
Looks good, but instead of coercing an int to a string we should change the key to an enum of `int | str` (and maybe more in future).

See https://github.com/pydantic/pydantic/issues/4675#issuecomment-1415443982

Effectively you need to `impl Hash` for the enum like [this](https://stackoverflow.com/a/69192884/949890).

Reply:
with that, you shouldn't need to `laxer_str` or `find_json_validator`